### PR TITLE
Youtube characters limit

### PR DESCRIPTION
**Pull request by Reettaphant**
Youtube has a character limit on it's tag list. Mam should make sure that youtube keywords don't go over this character limit so that we don't end up in a situation where updates to youtube are failing because we are exceeding this limit
screen shot 2017-10-19 at 08 59 43
.

### CODE REVIEW

File: public/video-ui/src/components/Video/Display.js [hunk 1]
```diff
@@ -110,6 +112,22 @@ class VideoDisplay extends React.Component {
...
 
+  validateYouTubeKeywords = youTubeKeywords => {
+    const charLimit = YouTubeKeywords.maxCharacters;
+    const numberOfChars = getYouTubeTagCharCount(youTubeKeywords);
+
+    if (numberOfChars > charLimit) {
+
+      return new FieldNotification(
+        'required',
+        `Maximum characters allowed in youTube tags is ${charLimit}.`,
```
Comment (gtrufitt) [inline public/video-ui/src/components/Video/Display.js]: Minor but `youTube` should be `YouTube`

File: public/video-ui/src/util/getYouTubeTagCharCount.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export function getYouTubeTagCharCount(tags) {
+  if (Array.isArray(tags)) {
+    return tags.reduce((charCount, keyword) => {
```
Comment (gtrufitt) [inline public/video-ui/src/util/getYouTubeTagCharCount.js]: Could be:
```js
return tags.reduce((charCount, keyword) => charCount += keyword.length, 0);
```

File: public/video-ui/src/components/FormFields/TagPicker.js [hunk 3]
```diff
@@ -330,6 +331,17 @@ export default class TagPicker extends React.Component {
     }
   }
 
+  renderCharCount() {
+    if (this.props.tagType === TagTypes.youtube) {
+      const charCount = getYouTubeTagCharCount(this.props.fieldValue);
+      return (
+        <span>
+          Character count: {charCount}
```
Comment (gtrufitt) [inline public/video-ui/src/components/FormFields/TagPicker.js]: Could you put `{charCount} / {YouTubeKeywords.maxCharacters}` so it's clearer what the limit is as the user is typing?

File: public/video-ui/src/components/FormFields/TagPicker.js [hunk 4]
```diff
@@ -392,6 +406,11 @@ export default class TagPicker extends React.Component {
           {this.props.notification.message}
           </p>
             : ''}
+        {hasError
```
Comment (gtrufitt) [inline public/video-ui/src/components/FormFields/TagPicker.js]: I wonder if using the logical `&&` here is cleaner than the ternary: 

```js
hasError && <p className="form__message form__message--error">
           {this.props.notification.message}
           </p>
```

Though it looks as if there's a consistency of using ternary operators with empty string in else, so not major

**Comment by RichieAHB** [comment]
Does the YT limit include separators too between keywords (i.e. the comma). If not then this is fine but if it does then you might want:

`return tags.reduce((charCount, keyword) => charCount += keyword.length, 0) + tags.length - 1`

**Comment by Reettaphant** [comment]
@RichieAHB Yes, separators are included in the character count.

File: public/video-ui/src/components/Video/Display.js (hunk 1 above)
Comment (Reettaphant) [reply public/video-ui/src/components/Video/Display.js]: 👍

File: public/video-ui/src/util/getYouTubeTagCharCount.js (hunk 2 above)
Comment (Reettaphant) [reply public/video-ui/src/util/getYouTubeTagCharCount.js]: 👍

File: public/video-ui/src/components/FormFields/TagPicker.js (hunk 4 above)
Comment (Reettaphant) [reply public/video-ui/src/components/FormFields/TagPicker.js]: I'll leave these alone for the reason you stated.

**Pull request merged**